js/interface.js

```diff
@@ -750,27 +750,30 @@ function save(origin, submission) {
             var cloneCredentialsPromise = Promise.resolve();
 
             newSubmission.data['fl-credentials'] = 'submission-' + newSubmission.id;
-
-            if (origin === "appStore") {
-              appStoreSubmission = newSubmission;
-              cloneCredentialsPromise = cloneCredentials(previousCredentials, appStoreSubmission);
-            } else if (origin === "enterprise") {
-              enterpriseSubmission = newSubmission;
-              cloneCredentialsPromise = cloneCredentials(previousCredentials, enterpriseSubmission);
-            } else if (origin === "unsigned") {
-              unsignedSubmission = newSubmission;
-              cloneCredentialsPromise = cloneCredentials(previousCredentials, unsignedSubmission);
-            }
-
-            return cloneCredentialsPromise.then(function () {
-              return Fliplet.App.Submissions.update(newSubmission.id, newSubmission.data);
-            }).then(function () {
-              $('.save-' + origin + '-progress').addClass('saved');
-
-              setTimeout(function () {
-                $('.save-' + origin + '-progress').removeClass('saved');
-              }, 4000);
-            });
+            
+            // Before we will decide clone credentials or not we should check has it been set by this time
+            return getCredential(previousCredentials).then(function(credentialIsExist) {
+              if (origin === "appStore" && credentialIsExist) {
+                appStoreSubmission = newSubmission;
+                cloneCredentialsPromise = cloneCredentials(previousCredentials, appStoreSubmission);
+              } else if (origin === "enterprise" && credentialIsExist) {
+                enterpriseSubmission = newSubmission;
+                cloneCredentialsPromise = cloneCredentials(previousCredentials, enterpriseSubmission);
+              } else if (origin === "unsigned" && credentialIsExist) {
+                unsignedSubmission = newSubmission;
+                cloneCredentialsPromise = cloneCredentials(previousCredentials, unsignedSubmission);
+              }
+  
+              return cloneCredentialsPromise.then(function () {
```

Comment:
can you also verify what happens when the clone fails? from my experience it completely breaks the UI. I think that when it fails it should just carry on as if there wasn't any credential set.

Reply:
@squallstar Yes you are right if this line ends with fail code will be stopped and UI will be broken.
With the current fix, if there is no credentials where set `cloneCredentialsPromise` will hold only the `Promise.resolve()` so it wouldn't fail now.

Reply:
great, thanks for confirming